### PR TITLE
WIP - Build and Packaging Updates

### PR DESCRIPTION
**Pull request by rossmcdonald**
- Fixes an issue with pre-installation script that can overwrite existing configurations.
- Re-enables ARM builds facilitated by the `build.py` script.
- Added optional `--goarm=` flag to `build.py` for customization around ARM version to target
- ~~Added basic Go bootstrapping retry logic when cross-compiling (the script will attempt to bootstrap Go for the target build).~~
- Small bug fixes around iterations and release candidates in `build.py`.

### CODE REVIEW

File: scripts/post-install.sh [hunk 1]
```diff
@@ -10,9 +10,9 @@ if ! id influxdb >/dev/null 2>&1; then
 fi
 chmod a+rX $BIN_DIR/influx*
 
-mkdir -p $LOG_DIR
+test -d $LOG_DIR || mkdir -p $LOG_DIR
```
Comment (otoolep) [inline scripts/post-install.sh]: mkdir was causing an error?

**Comment by otoolep** [comment]
Install script changes seem fine to me.

File: scripts/post-install.sh (hunk 1 above)
Comment (rossmcdonald) [inline scripts/post-install.sh]: Nothing serious, but in some of my testing I saw a few `mkdir: cannot create directory` errors which I assumed was due to the directories already existing.

**Comment by rossmcdonald** [comment]
@nathanielc Do you mind reviewing these changes? In addition to the items listed in the description of the PR:

- The package iteration is set to 0 when also specifying a release candidate
- Debian packages are now assigned an iteration

I think this is ready to merge as long as you don't have any concerns.

File: build.py [hunk 2]
```diff
@@ -330,7 +337,26 @@ def build(version=None,
             build_command += "-X main.branch={} ".format(branch)
             build_command += "-X main.commit={}\" ".format(get_current_commit())
         build_command += c
-        out = run(build_command, shell=True)
+        
+        try_bootstrap = False
```
Comment (nathanielc) [inline build.py]: Bootstrapping shouldn't be necessary, I would prefer it to just error instead of attempt to make a change to the system. We can do the bootstrapping in docker and then this script will just fail if its not setup correctly. Then we will just need to go fix the Dockerfile.

**Comment by nathanielc** [comment]
This looks good. +1 

The bootstrapping thing is just my preference. To me build.py should just build the code, if its deps are there it fails. Then we use docker to ensure it has its needed deps. For example build.py doesn't try to install fpm if it can't find it.

**Comment by rossmcdonald** [comment]
@nathanielc Thanks! I agree about the bootstrapping, so I've stripped it with the most recent commit. Since we've moved to Docker (and Go 1.5), we can just rely on that.

**Comment by nathanielc** [comment]
:shipit:

**Comment by rossmcdonald** [comment]
@nathanielc Do you mind reviewing this one again when you get a chance? Nothing has really changed since the last review except for the pre and post installation scripts. There is an issue where the different order of operations that `yum` and `apt-get` use to upgrade packages that will cause the init scripts to be removed on CentOS, so I added some distribution-specific logic to account for that.

File: scripts/pre-install.sh [hunk 3]
```diff
@@ -1,10 +1,16 @@
...
-fi
+if [[ -d /etc/opt/influxdb ]]; then
+    # Legacy configuration found
+    if [[ ! -d /etc/influxdb ]]; then
+	# New configuration does not exist, move legacy configuration to new location
+	echo -e "Please note, InfluxDB's configuration is now located at '/etc/influxdb' (previously '/etc/opt/influxdb')."
+	mv -vn /etc/opt/influxdb /etc/influxdb
 
-exit
+	if [[ -f /etc/influxdb/influxdb.conf ]]; then
```
Comment (nathanielc) [inline scripts/pre-install.sh]: Won't this always be false since above the directory doesn't even exist?

Also why is this necessary, both apt and yum will back up the config for us or am I missing something?

Comment (nathanielc) [inline scripts/pre-install.sh]: To answer to my first question: we potentially just moved the file there.

Is the backup just a safety thing since we just moved files on the package manager?

Comment (rossmcdonald) [inline scripts/pre-install.sh]: Yes, that is just to make 100% sure we have a backup. I've encountered a few issues with rpm where it won't save it correctly because the configuration file changed locations. In those instances, it creates an `influxdb.conf.$(date)~` backup, but it's just a copy of the vanilla configuration (without any customizations).

Comment (rossmcdonald) [inline scripts/pre-install.sh]: I'd be happy to demo this, btw. I tried all sorts of methods to prevent it, but rpm overwrites the configuration every-friggin-time.

Comment (nathanielc) [inline scripts/pre-install.sh]: Is that every-time we move the config during the config dir change or every-time even without changing the config dir?

Comment (rossmcdonald) [inline scripts/pre-install.sh]: It's every time when we move the config directory. If we don't move it, it creates an `influxdb.conf.rpmsave` in the `/etc/opt/influxdb/` directory instead. This issue doesn't happen with apt-get either, just rpm.

Comment (nathanielc) [inline scripts/pre-install.sh]: OK, I can see how rpm could get confused if we move things around during a pre-install step. Just glad its not broken for the basic case too. 

+1 this looks good to me.
